### PR TITLE
Change string formatting throughout package to use f-strings where appropriate

### PR DESCRIPTION
**Pull request by jorobert**
Pylint's standard is now to use f-strings rather than other ways of string formatting, primarily for efficiency and ease of reading.  I've adapted all of these where appropriate and put in pylint exceptions, primarily for some strange tuple unpacking where it's just cleaner not to.  There should be no change in behavior otherwise.  

Also addressed a few other Pylint warnings where we didn't declare file encoding.  I set these all to utf-8, assuming that's correct.

### CODE REVIEW

File: spotfire/spk.py [hunk 1]
```diff
@@ -838,7 +838,7 @@ def _handle_versioning(package_builder, installed_packages, brand, version, forc
         package_builder.version = given_version
     # Handle versioned filenames
     if versioned_filename:
-        package_builder.output = re.sub(r"(\.spk)?$", r"-%s\1" % str(package_builder.version),
+        package_builder.output = re.sub(r"(\.spk)?$", f"-{str(package_builder.version)}\1",
```
Comment (bbassett-tibco) [inline spotfire/spk.py:841]: Minor point: you can combine the `f` and `r` string modifiers.  In this case it actually has an impact on the regexp backreference (changing it to an ASCII 1 character):

```python
>>> r"\1"
'\\1'
>>> f"\1"
'\x01'
>>> fr"\1"
'\\1'
```

File: spotfire/data_function.py [hunk 2]
```diff
@@ -302,27 +302,24 @@ def _compile_script(self, result: AnalyticResult) -> None:
 
     def _read_inputs(self, result: AnalyticResult) -> None:
         """read inputs"""
-        self.debug("reading %d input variables" % len(self.inputs))
+        self.debug(f"reading {len(self.inputs)} input variables")
         for i, input_ in enumerate(self.inputs):
             if _bad_string(input_.name) or input_.name == "":
-                self.debug("input %d: bad input variable name - skipped" % i)
+                self.debug(f"input {int(i)}: bad input variable name - skipped")
```
Comment (bbassett-tibco) [inline spotfire/data_function.py:308]: Probably don't need to convert an int to int before stringing it as part of the f-string.

File: spotfire/data_function.py [hunk 3]
```diff
@@ -350,23 +347,23 @@ def _execute_script(self, compiled_script: typing.Any, result: AnalyticResult) -
 
     def _write_outputs(self, result: AnalyticResult) -> None:
         """write outputs"""
-        self.debug("writing %d output variables" % len(self.outputs))
+        self.debug(f"writing {len(self.outputs)} output variables")
         for i, output in enumerate(self.outputs):
             if _bad_string(output.name) or output.name == "":
-                self.debug("output %d: bad output variable name - skipped" % i)
+                self.debug(f"output {int(i)}: bad output variable name - skipped")
```
Comment (bbassett-tibco) [inline spotfire/data_function.py:353]: Again, int'ing an int isn't probably needed.

File: spotfire/data_function.py [hunk 4]
```diff
@@ -85,12 +85,12 @@ def read(self, globals_dict: typing.Dict[str, typing.Any], debug_fn: typing.Call
...
         if self.type == "NULL":
-            debug_fn("assigning missing '%s' as None" % self.name)
+            debug_fn(f"assigning missing '{self.name}' as None")
             globals_dict[self.name] = None
             return
-        debug_fn("assigning %s '%s' from file %s" % (self.type, self.name, self.file))
+        debug_fn(f"assigning {self.type} '{self.name}' from file {self.file}")
         dataframe = sbdf.import_data(self.file)
-        debug_fn("read %d rows %d columns" % (dataframe.shape[0], dataframe.shape[1]))
+        debug_fn(f"read {int(dataframe.shape[0])} rows {int(dataframe.shape[1])} columns")
```
Comment (bbassett-tibco) [inline spotfire/data_function.py:93]: Another unnecessary `int()` call.

File: spotfire/sbdf.py [hunk 5]
```diff
@@ -64,7 +64,7 @@ def import_data(sbdf_file: typing.Union[str, bytes, int]) -> pd.DataFrame:
         # Read the file header
         version_major, version_minor = _FileHeader.read(file)
         if version_major != _FileHeader.Major_Version or version_minor != _FileHeader.Minor_Version:
-            raise SBDFError("unsupported file version %d.%d" % (version_major, version_minor))
+            raise SBDFError(f"unsupported file version {int(version_major)}.{int(version_minor)}")
```
Comment (bbassett-tibco) [inline spotfire/sbdf.py:67]: Another unnecessary `int()` call.

File: spotfire/test/utils.py [hunk 6]
```diff
@@ -37,4 +37,4 @@ def generate_reports(self, test_runner):
         test_runner.output.write(xml_content)
 
 
-PYTHON_VERSION = "py%d%d" % (sys.version_info.major, sys.version_info.minor)
+PYTHON_VERSION = f"py{int(sys.version_info.major)}{int(sys.version_info.minor)}"
```
Comment (bbassett-tibco) [inline spotfire/test/utils.py:40]: Another unnecessary `int()` call.

**Review by bbassett-tibco — CHANGES_REQUESTED** [review]
Looks mostly good; nothing problematic.  Just a few minor comments sprinkled throughout.

File: spotfire/data_function.py (hunk 2 above)
Comment (jorobert) [reply spotfire/data_function.py:308]: Good catches!  Yeah, the flynt package I used to do the mostly automatic conversions doesn't know when %d requires an explicit cast to int or not.  Thanks for catching these.  I'll adjust.

File: spotfire/spk.py (hunk 1 above)
Comment (jorobert) [reply spotfire/spk.py]: Good catch!  Updated to use "fr" string instead of just "f" string.

**Review by bbassett-tibco — APPROVED** [review]
LGTM.

**Pull request closed without merging**